contracts/Core.sol

```diff
@@ -60,17 +84,26 @@ contract Core is CoreInterface {
 		address _registrar,
 		uint256 _chainIdOrigin,
 		uint256 _chainIdRemote,
-		address _openSTRemote)
+		address _openSTRemote,
+		WorkersInterface _workers)
 		public
 	{
-		require(_registrar != address(0));
-		require(_chainIdOrigin != 0);
-		require(_chainIdRemote != 0);
-		require(_openSTRemote != 0);
+		require(_registrar != address(0), "Registrar should be valid address");
+		require(_chainIdOrigin != 0, "Invalid origin chain ID");
+		require(_chainIdRemote != 0, "Invalid remote chain ID");
+		require(_openSTRemote != 0, "Invalid openSTRemote contract address");
+		require(_workers != address(0), "Workers should be valid contract address");
 		coreRegistrar = _registrar;
 		coreChainIdOrigin = _chainIdOrigin;
 		coreChainIdRemote = _chainIdRemote;
+		// TODO - openSTRemote is openSTValue and openSTUtility contract address?
 		coreOpenSTRemote = _openSTRemote;
+		workers = _workers;
```

Comment:
openSTRemote is openSTValue and openSTUtility contract address on opposite chains. Correct?